File: overview/src/main/java/cn/edu/nuc/overview/OverviewFragment.java
package cn.edu.nuc.overview;

import android.os.Bundle;
import android.view.View;
import android.webkit.WebView;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.zhouwei.mzbanner.MZBannerView;
import com.zhouwei.mzbanner.holder.MZHolderCreator;
import com.zhouwei.mzbanner.holder.MZViewHolder;

import java.util.Arrays;
import java.util.List;

import cn.edu.nuc.library.base.BaseFragment;

/**
 * Author: Inno Fang
 * Time: 2017/8/19 12:26
 * Description:
 */

@Route(path = "/pbs_ov_fragment/1")
public class OverviewFragment extends BaseFragment {

    private MZBannerView<Integer> mBanner;
    private WebView mWebView;

    public static OverviewFragment newInstance() {
        return new OverviewFragment();
    }

    @Override
    protected int getLayoutResId() {
        return R.layout.ov_fragment;
    }

    @Override
    protected void createView(View view, Bundle savedInstanceState) {
        mBanner = (MZBannerView<Integer>) find(R.id.banner);
        mWebView = (WebView) find(R.id.content_web_view);
    }

    @Override
    protected void initEvent() {

        List<Integer> res = Arrays.asList(
                R.drawable.banner_1,
                R.drawable.banner_2,
                R.drawable.banner_3,
                R.drawable.banner_4,
                R.drawable.banner_5);

        mBanner.setBannerPageClickListener(new MZBannerView.BannerPageClickListener() {
            @Override
            public void onPageClick(View view, int i) {

            }
        });

        mBanner.setPages(res, new MZHolderCreator() {
            @Override
            public MZViewHolder createViewHolder() {
                return new BannerViewHolder();
            }
        });

        mWebView.getSettings().setJavaScriptEnabled(true);
        mWebView.loadUrl("file:///android_asset/ov_index.html");
    }

    @Override
    public void onPause() {
        super.onPause();
        mBanner.pause();
    }

    @Override
    public void onResume() {
        super.onResume();
        mBanner.start();
    }
}

File: overview/src/main/java/cn/edu/nuc/overview/OverviewFragment.kt
package cn.edu.nuc.overview

import android.os.Bundle
import android.webkit.WebView
import cn.edu.nuc.library.base.BaseFragment
import cn.edu.nuc.library.extensions.toast
import com.zhouwei.mzbanner.MZBannerView

/**
 * Author: Inno Fang
 * Time: 2017/8/18 12:55
 * Description:
 */

class OverviewFragment : BaseFragment() {

    companion object {
        fun newInstance(): OverviewFragment = OverviewFragment()
    }

    private val res = intArrayOf(
            R.drawable.banner_1,
            R.drawable.banner_2,
            R.drawable.banner_3,
            R.drawable.banner_4,
            R.drawable.banner_5)

    private val banner by lazy { find<MZBannerView<Int>>(R.id.banner) }
    private val wvContent by lazy { find<WebView>(R.id.content_web_view) }

    override val layoutResId: Int = R.layout.ov_fragment

    override fun createView(savedInstanceState: Bundle?) {
        banner.setBannerPageClickListener { _, pos -> toast("click page $pos") }
        banner.setPages(res.toMutableList()) { BannerViewHolder() }

        wvContent.settings.javaScriptEnabled = true
        wvContent.loadUrl("file:///android_asset/ov_index.html")
    }

    override fun onPause() {
        super.onPause()
        banner.pause()
    }

    override fun onResume() {
        super.onResume()
        banner.start()
    }

}